Refined dbfs implementation and tested it in a real cluster

nils-braun opened the pull request:
Thanks for the nice start @martindurant! 
I did some small refactorings and cleanings, fixed some small bugs and tested it on a "real" Azure databricks cluster. I can list, create and read files :-) 

Still to do:
* fix the open TODO
* add tests
* add documentation
* add handling for errors, exceptions or edge-cases

martindurant commented on fsspec/implementations/dbfs.py:
```diff
@@ -1,66 +1,115 @@
+        return base64.b64decode(r["data"])
+
+    def post_to_api(self, endpoint, json):
+        r = self.session.post(os.path.join(self.url, "2.0/dbfs", endpoint), json=json)
```
os.path.join will do the wrong thing on windows. urlparse has a join, but I think "/" or just `+` are fine.

martindurant commented on fsspec/implementations/dbfs.py:
```diff
@@ -1,66 +1,115 @@
+        return r.json()
 
-class DBFile(AbstractBufferedFile):
+    def get_from_api(self, endpoint, json):
```
get and post are similar enough, I would make join them, with a `method=` kwarg

martindurant commented on fsspec/implementations/dbfs.py:
```diff
@@ -1,66 +1,115 @@
+            json={"source_path": path1, "destination_path": path2},
+        )
+
+    def create_handle(self, path, overwrite=True):
```
These nicely reflect the real API, but are not meant to be called directly by users, so might prefix with underscore

martindurant commented on fsspec/implementations/dbfs.py:
```diff
@@ -86,23 +143,17 @@ def __init__(
-                                     json={"handle": self.path})
-            r.raise_for_status
+            print("Closing handle", self.handle)
+            self.fs.close_handle(handle=self.handle)
```
Should return `True` to clear the write buffer

martindurant reviewed (commented):
Some comments so far.
Thanks for the input!

nils-braun commented:
Thanks for the comments, @martindurant!
The print statements do also need to go away of course ;-)

nils-braun commented:
I have implemented your proposals and added some more documentation and error handling (handling all the exceptions the API documentation specifies).
The code now also works with large file uploads and downloads. 

The only thing missing are now the tests, but we might want to do this in a second PR.

martindurant commented:
We should probably have at least a very minimal test first, and then we can iterate.

martindurant commented:
I made you a collaborator on this fork, so you should be able to push to my branch, if you want @nils-braun
